src/mad_spark_alt/evolution/fitness.py

```
@@ -232,45 +294,23 @@
+            
+            # Try fallback if available
+            if self.fallback_diversity_calculator:
+                try:
+                    diversity = await self.fallback_diversity_calculator.calculate_diversity(population)
+                    logger.info("Using fallback diversity calculator")
+                    return max(0.0, min(1.0, diversity))
+                except Exception as fallback_e:
+                    logger.error(f"Fallback diversity calculator also failed: {fallback_e}")
+            
+            # If all fails, return default
+            logger.error(f"All diversity calculators failed, using default score")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)